src/main/java/net/sf/jabref/model/database/EntrySorter.java

```diff
@@ -90,8 +90,10 @@ public void databaseChanged(DatabaseChangeEvent e) {
             int pos;
             switch (e.getType()) {
             case ADDED_ENTRY:
-                pos = -Collections.binarySearch(set, e.getEntry(), comp) - 1;
-                set.add(pos, e.getEntry());
+                pos = Collections.binarySearch(set, e.getEntry(), comp);
```

Comment:
This relates to a rather strange thing that I do not really understand. When I played around with changing the file in a separate editor to trigger change detection, I found out that lead to the fix in EntryChange. After fixing that, the detection code behaved differently (there's a potential reason though). Before, removing a field leads to a changed entry. After, removing a field leads to a removed and an added entry. I *think* it depends on which field is removed since duplicate detection is used to determine if the entry is similar enough to be considered as modified or two completely different entries.

To answer your question, I think this makes more sense. If an ADDED_ENTRY event is triggered, it is checked if the entry already exists. If not, it is added. The problem with the previous code was that it only worked if the entry didn't exist (otherwise pos would be negative and an exception was thrown). I guess a fundamental reason behind the problem here is that similarity is checked with duplicate detection in the other place and a comparator here. Note that the only time the entry is not added is when it already exists...